src/main/java/com/unstoppabledomains/resolution/contracts/Contract.java

```diff
@@ -0,0 +1,58 @@
+package com.unstoppabledomains.resolution.contracts;
+
+import java.io.FileReader;
+import java.io.IOException;
+
+import com.unstoppabledomains.exceptions.NSExceptionCode;
+import com.unstoppabledomains.exceptions.NSExceptionParams;
+import com.unstoppabledomains.exceptions.NamingServiceException;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class Contract {
+  protected String address;
+  protected String type;
+  protected JSONArray abi;
+
+  public Contract(String address, String type, String pathToAbi) throws NamingServiceException {
+    JSONParser parser = new JSONParser();
+    try {
+      String jsonAbiPath = type.equals("registry") 
+        ? pathToAbi + "/registry.json"
+        : pathToAbi + "/`resolver.json";
+      System.out.println(jsonAbiPath);
+      JSONArray parsed = (JSONArray) parser.parse(new FileReader(jsonAbiPath));
+      this.abi = parsed;
+    } catch (IOException | ParseException e) {
+      throw new NamingServiceException(NSExceptionCode.UnknownError, null, e);
+    }
+    if (address == null || address.isEmpty())
+      throw new NamingServiceException(NSExceptionCode.IncorrectContractAddress,
+          new NSExceptionParams("a", address));
```

Comment:
I remember we were throwing `Domain is not configured` exception when there were no resolver for domain. I think we need it here too.

Reply:
This is more of a developer's eyes only error. The address here is the contract address. I was planning to capture that error above and throw Domain "unconfigured.crypto" is not configured caused by IncorrectContractAddress ^_^ error chainlink indeed is comfortable thing =) 

should i thow a regular exception instead of NamingServiceException for such case?

Reply:
Yes, I think so: `NamingServiceException` is user facing exceptions and should only have codes that need to be processed by the end user.